### PR TITLE
Removed the v6 incompatible subscribe messages

### PR DESCRIPTION
**Pull request by SzymonPobiega**
to prevent potential message loss

### CODE REVIEW

**Comment by danielmarbach** [comment]
@SzymonPobiega needs a rebase

File: src/NServiceBus.AcceptanceTests/Msmq/When_subscribing_from_a_worker.cs [hunk 1]
```diff
@@ -3,32 +3,33 @@
...
 
-    public class When_publishing_to_a_legacy_endpoint : NServiceBusAcceptanceTest
+    public class When_subscribing_from_a_worker : NServiceBusAcceptanceTest
     {
         [Test]
-        public async Task Should_be_delivered()
+        public async Task Event_should_be_delivered_to_the_distributor()
         {
-            await Scenario.Define<Context>()
+           var ctx = await Scenario.Define<Context>()
```
Comment (danielmarbach) [inline src/NServiceBus.AcceptanceTests/Msmq/When_subscribing_from_a_worker.cs]: We call this one mostly context

File: src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj [hunk 2]
```diff
@@ -144,16 +144,16 @@
...
       <SubType>Code</SubType>
     </Compile>
+    <Compile Include="Msmq\When_subscribing_from_a_worker.cs" />
     <Compile Include="Routing\When_publishing_to_scaled_out_subscribers_on_unicast_transports.cs" />
     <Compile Include="Routing\When_publishing_to_scaled_out_subscribers_on_multicast_transports.cs" />
     <Compile Include="Msmq\When_publishing_with_authorizer.cs" />
     <Compile Include="Msmq\When_unsubscribing_with_authorizer.cs" />
     <Compile Include="Routing\When_broadcasting_a_command.cs" />
-    <Compile Include="Routing\When_replying_to_a_message_sent_via_a_distributor.cs" />
+    <Compile Include="Msmq\When_replying_to_a_message_sent_via_a_distributor.cs" />
```
Comment (danielmarbach) [inline src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj]: So the test belongs now to MSMQ ATT proj

File: src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj [hunk 3]
```diff
@@ -144,16 +144,16 @@
     <Compile Include="Routing\AutomaticSubscriptions\When_starting_an_endpoint_with_autoSubscribe.cs">
       <SubType>Code</SubType>
     </Compile>
+    <Compile Include="Msmq\When_subscribing_from_a_worker.cs" />
```
Comment (danielmarbach) [inline src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj]: So the test belongs now to MSMQ ATT proj

File: src/NServiceBus.AcceptanceTests/Msmq/When_subscribing_from_a_worker.cs (hunk 1 above)
Comment (SzymonPobiega) [inline src/NServiceBus.AcceptanceTests/Msmq/When_subscribing_from_a_worker.cs]: :+1:

File: src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj (hunk 3 above)
Comment (SzymonPobiega) [inline src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj]: MSMQ ATT proj?

Comment (SzymonPobiega) [inline src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj]: :+1:

File: src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj (hunk 2 above)
Comment (SzymonPobiega) [inline src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj]: :+1:

**Comment by SzymonPobiega** [comment]
@danielmarbach rebased, squashed and implemented your comments.

File: src/NServiceBus.Msmq.AcceptanceTests/When_subscribing_from_a_worker.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                public Task Handle(MyEvent messageThatIsEnlisted, IMessageHandlerContext context)
+                {
+                    Context.DeliveredToWorker = true;
+
+                    return Task.FromResult(0);
+                }
+            }
+        }
+
+        [Serializable]
```
Comment (andreasohlund) [inline src/NServiceBus.Msmq.AcceptanceTests/When_subscribing_from_a_worker.cs]: Redundant

File: src/NServiceBus.Msmq.AcceptanceTests/When_subscribing_from_a_worker.cs [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+        public class Worker : EndpointConfigurationBuilder
+        {
+            public Worker()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.EnlistWithLegacyMSMQDistributor(DistributorEndpoint, DistributorEndpoint, 1);
+                })
+                    .AddMapping<MyEvent>(typeof(Publisher));
```
Comment (danielmarbach) [inline src/NServiceBus.Msmq.AcceptanceTests/When_subscribing_from_a_worker.cs]: Weird spacing

File: src/NServiceBus.Msmq.AcceptanceTests/When_subscribing_from_a_worker.cs (hunk 4 above)
Comment (danielmarbach) [inline src/NServiceBus.Msmq.AcceptanceTests/When_subscribing_from_a_worker.cs]: What the master said ;)

**Comment by SzymonPobiega** [comment]
Fixed, your grumpiness

**Pull request merged**